examples/platforms/cc2650/radio.c

```diff
@@ -0,0 +1,1696 @@
+    sRadioSetupCmd.pRegOverride = sIEEEOverrides;
+    sRadioSetupCmd.mode = 1;
+
+    interruptsWereDisabled = IntMasterDisable();
+
+    if ((ret = (RFCDoorbellSendTo((uint32_t)&sStartRatCmd) & 0xFF)) != CMDSTA_Done)
+    {
+        if (!interruptsWereDisabled)
+        {
+            IntMasterEnable();
+        }
+
+        return ret;
+    }
+
+    /* synchronously wait for the CM0 to stop executing */
+    while ((HWREG(RFC_DBELL_NONBUF_BASE + RFC_DBELL_O_RFCPEIFG) & IRQ_LAST_COMMAND_DONE) == 0x00);
+
+    if (!interruptsWereDisabled)
+    {
+        IntMasterEnable();
+    }
+
+    return sRadioSetupCmd.status;
+}
+
+/**
+ * @brief sends the shutdown command string to the radio core
+ *
+ * Powers down the frequency synthesizer and stops the RAT.
+ *
+ * @note synchronously waits until the command string completes.
+ *
+ * @return the status of the RAT stop command
+ * @retval DONE_OK the command string executed properly
+ */
+static uint_fast16_t rfCoreSendDisableCmd(void)
+{
+    uint8_t doorbell_ret;
+    bool interruptsWereDisabled;
+
+    HWREGBITW(AON_RTC_BASE + AON_RTC_O_CTL, AON_RTC_CTL_RTC_UPD_EN_BITN) = 1;
+
+    /* initialize the command to power down the frequency synth */
+    memset((void *)&sFsPowerdownCmd, 0, sizeof(sFsPowerdownCmd));
+    sFsPowerdownCmd.commandNo = CMD_FS_POWERDOWN;
+    sFsPowerdownCmd.condition.rule = COND_ALWAYS;
+    sFsPowerdownCmd.pNextOp = (rfc_radioOp_t *)&sStopRatCmd;
+
+    memset((void *)&sStopRatCmd, 0, sizeof(sStopRatCmd));
+    sStopRatCmd.commandNo = CMD_SYNC_STOP_RAT;
+    sStopRatCmd.condition.rule = COND_NEVER;
+
+    interruptsWereDisabled = IntMasterDisable();
+
+    HWREG(RFC_DBELL_NONBUF_BASE + RFC_DBELL_O_RFCPEIFG) = ~IRQ_LAST_COMMAND_DONE;
+
+    doorbell_ret = (RFCDoorbellSendTo((uint32_t)&sFsPowerdownCmd) & 0xFF);
+
+    if (doorbell_ret != CMDSTA_Done)
+    {
+        if (!interruptsWereDisabled)
+        {
+            IntMasterEnable();
+        }
+
+        return doorbell_ret;
+    }
+
+    /* synchronously wait for the CM0 to stop */
+    while ((HWREG(RFC_DBELL_NONBUF_BASE + RFC_DBELL_O_RFCPEIFG) & IRQ_LAST_COMMAND_DONE) == 0x00);
+
+    if (!interruptsWereDisabled)
+    {
+        IntMasterEnable();
+    }
+
+    if (sStopRatCmd.status == DONE_OK)
+    {
+        sRatOffset = sStopRatCmd.rat0;
+    }
+
+    return sStopRatCmd.status;
+}
+
+/**
+ * error interrupt handler
+ */
+void RFCCPE1IntHandler(void)
+{
+    if (!PRCMRfReady())
+    {
+    }
+
+    /* Clear INTERNAL_ERROR interrupt flag */
+    HWREG(RFC_DBELL_NONBUF_BASE + RFC_DBELL_O_RFCPEIFG) = 0x7FFFFFFF;
+}
+
+/**
+ * command done handler
+ */
+void RFCCPE0IntHandler(void)
+{
+    if (HWREG(RFC_DBELL_NONBUF_BASE + RFC_DBELL_O_RFCPEIFG) & IRQ_LAST_COMMAND_DONE)
+    {
+        HWREG(RFC_DBELL_NONBUF_BASE + RFC_DBELL_O_RFCPEIFG) = ~IRQ_LAST_COMMAND_DONE;
+
+        if (sState == cc2650_stateReceive)
+        {
+            if (sReceiveCmd.status != ACTIVE && sReceiveCmd.status != IEEE_SUSPENDED)
```

Comment:
It can be, the outer `if` statement was separate because it was in a `else if` chain of states.

This interrupt handler has gone through a few iterations, I'll try to make it more readable.